src/global/Pattern.h

```diff
@@ -276,16 +280,43 @@ CompactVectorOfStrings<DataT>::diskIterator(string filename) {
   size_t offset;
   indexFile.read(&offset, sizeof(offset));
 
-  for (size_t i = 0; i < size; ++i) {
-    size_t nextOffset;
-    indexFile.read(&nextOffset, sizeof(nextOffset));
-    auto currentSize = nextOffset - offset;
-    vector_type result;
-    result.resize(currentSize);
-    dataFile.read(result.data(), currentSize * sizeof(DataT));
-    co_yield result;
-    offset = nextOffset;
-  }
+  struct Generator : public ad_utility::InputRangeFromGet<
+                         typename CompactVectorOfStrings<DataT>::vector_type> {
+    explicit Generator(ad_utility::File&& dataFile,
+                       ad_utility::File&& indexFile, const std::size_t a_size,
+                       const std::size_t an_offset)
+        : data{std::move(dataFile)},
+          index{std::move(indexFile)},
+          size{a_size},
+          offset{an_offset} {}
```

Comment:
I was first thinking of some ways to simplify this
(CachingTransformInputRangeFromGet, using a lambda), 
but as it seems, this function is completely unused (other than in its unit tests), so feel free to just remove it. 
This simple change can also be in a separate small PR to make the revewing process more efficient.

Reply:
please let me resolve this issue as the relevant pr i merged.